ironfish/src/migrations/data/013-accounts-dedupe.ts

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { IronfishNode } from '../../node'
+import { IDatabase } from '../../storage'
+import { createDB } from '../../storage/utils'
+import { Migration } from '../migration'
+
+export class Migration013 extends Migration {
+  path = __filename
+
+  async prepare(node: IronfishNode): Promise<IDatabase> {
+    await node.files.mkdir(node.accounts.db.location, { recursive: true })
+    return createDB({ location: node.accounts.db.location })
+  }
+
+  async forward(node: IronfishNode): Promise<void> {
+    const seenKeys = new Set()
+    const duplicateAccounts = []
+
+    for (const account of node.accounts.listAccounts()) {
+      if (seenKeys.has(account.spendingKey)) {
+        duplicateAccounts.push(account)
+      } else {
+        seenKeys.add(account.spendingKey)
+      }
+    }
+
+    for (const duplicateAccount of duplicateAccounts) {
+      await node.accounts.removeAccount(duplicateAccount.name)
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  async backward(): Promise<void> {}
+}
```

Comment:
I assume the reason you want to do this is to make the data consistent with the variant right? That way no one ever tries to map on the spend key? I think it's probably ok to allow this for now and get this change out without the migration since it already improves things for users.

Reply:
Well, we just shouldn't allow it because like you're alluding to, they can get out of sync and just cause issues.

But I'm mostly concerned about adding the ability to stop trying to decrypt a note as soon as one account has decrypted it. This should lead to fairly substantial speed increases for users with multiple accounts, but we can't do that until we can guarantee account uniqueness.

Reply:
Ok, what about this...

1. Merge the change on its own because I think it's a good improvement
2. Merge in the migration along with wallet 2.0 so users run it anyway when running wallet 2.0 migrations and they don't get hit with more migrations
3. Open PR with optimization now that we know wallets are unique per spend key.